docs_src/design/adr/0019-Service-List.md

```diff
@@ -0,0 +1,92 @@
+  - Services would have to monitor the file or volume for changes in order to be able to address the add/removal of services at runtime.
+  - How would the security bootstrapping address the add/removal of services and get new tokens established?  The bootstrapping service isn’t even long running to do this.
+- Have global configuration in Consul (or config/reg service) that everyone reads.
+  - When Consul is used, there would be a single authoritative list of services provided to all requesting services through Consul. Updates to Consul would be transactional (ACID).
+  - When Consul is not used, have a common service list in each service configuration.toml file that everyone reads.  Further, when Consul is not used, each service is using the same means (addressing #1 above) to get its service list (that is going to its config TOML file locally).  However, each service list in each configuration file could be different (leading to service list inconsistencies) and there is not a single, authoritative list when using local configuration (making ACID transactions to the service list per #3 above very difficult – especially if services and their configuration file holding the copy of the service list are distributed across hosts).
+  - Using Consul would make sure that the service list is well formed (meaining adherence to format, but this does not mean that the service list necessarily adheres to any validity rules unless additional code interacting with Consul changes is checking for well formed-ness and validity) and that any change operation is ACID (but only with regard to Consul and the service list - this does not mean the transaction would incorporate additional needs such as the API Gateway updating new routes).
+  - When using the local configuration, some of the same issues exist as listed in the first alternative (validation of the file, etc.).
+  - Services would have to be notified by Consul if the service list changes or monitor the local config for a service list change in order to be able to address the add/removal of services at runtime.
+  - Again, how would the security bootstrapping address the add/removal of services and get new tokens established?  The bootstrapping service isn’t even long running to do this.
+  - We could use Consul for dynamic service needs and when Consul is not used resolve that the service list is static and requires a restart of the services.
+  - There is a mutually inclusive event issue (aka chicken and egg problem) with this solution in that we have to secure Consul via the security bootstrapping providing it a token and a list of all the services, but Consul is the place where the security service would get the list of all services.  The circumstances prohibit both to be true.
+- Use environment variables to provide the list of services.
+  - Each micro service would use the same env var to get the list of services.
```

Comment:
Right, but you can't really do this. They can use the same env var name, but the actually value is a per-process thing, unless you use an OS/orchestration mechanism to ensure that the value is the same for all processes...